javaslang-benchmark/src/test/java/javaslang/benchmark/collection/TreeMapBenchmark.java

```diff
@@ -0,0 +1,398 @@
+            keysRandom = null;
+            keysRandom = null;
+            entriesRandom = null;
+            slangTreeMap = null;
+            javaTreeMap = null;
+            scalaTreeMap = null;
+        }
+
+        // Valid keys are only even numbers
+        int nextValidRandomKey() {
+            return keysRandom.get(random.nextInt(numItems));
+        }
+
+        int nextRandomKey() {
+            return random.nextInt(maxKey);
+        }
+
+        Array<Integer> randomize(Array<Integer> list) {
+            java.util.List<Integer> integers = list.toJavaList();
+            Collections.shuffle(integers);
+            return Array.ofAll(integers);
+        }
+    }
+
+    @Benchmark
+    @Group("GetAndPut")
+    public Object putRandomValues_Slang(BenchmarkState state) {
+        TreeMap<Integer, String> treeMap = TreeMap.empty();
+        for (Integer key : state.keysRandom) {
+            treeMap = treeMap.put(key, Integer.toString(key));
+        }
+        return treeMap;
+    }
+
+    @Benchmark
+    @Group("GetAndPut")
+    public Object putRandomValues_Java(BenchmarkState state) {
+        java.util.TreeMap<Integer, String> treeMap = new java.util.TreeMap<>();
+        for (Integer key : state.keysRandom) {
+            treeMap.put(key, Integer.toString(key));
+        }
+        return treeMap;
+    }
+
+    @Benchmark
+    @Group("GetAndPut")
+    public Object putRandomValues_Scala(BenchmarkState state) {
+        scala.collection.immutable.TreeMap<Integer, String> treeMap = new scala.collection.immutable.TreeMap<>(SCALA_ORDERING);
+        for (Integer key : state.keysRandom) {
+            treeMap = treeMap.$plus(new scala.Tuple2<>(key, Integer.toString(key)));
+        }
+        return treeMap;
+    }
+
+    @Benchmark
+    @Group("GetAndPut")
+    public Object putSortedValues_Slang(BenchmarkState state) {
+        TreeMap<Integer, String> treeMap = TreeMap.empty();
+        for (Integer key : state.keysSorted) {
+            treeMap = treeMap.put(key, Integer.toString(key));
+        }
+        return treeMap;
+    }
+
+    @Benchmark
+    @Group("GetAndPut")
+    public Object putSortedValues_Java(BenchmarkState state) {
+        java.util.TreeMap<Integer, String> treeMap = new java.util.TreeMap<>();
+        for (Integer key : state.keysSorted) {
+            treeMap.put(key, Integer.toString(key));
+        }
+        return treeMap;
+    }
+
+    @Benchmark
+    @Group("GetAndPut")
+    public Object putSortedValues_Scala(BenchmarkState state) {
+        scala.collection.immutable.TreeMap<Integer, String> treeMap = new scala.collection.immutable.TreeMap<>(SCALA_ORDERING);
+        for (Integer key : state.keysSorted) {
+            treeMap = treeMap.$plus(new scala.Tuple2<>(key, Integer.toString(key)));
+        }
+        return treeMap;
+    }
+
+    @Benchmark
+    @Group("GetAndPut")
+    public Object getValue_Slang(BenchmarkState state) {
+        return state.slangTreeMap.get(state.nextValidRandomKey());
```

Comment:
Ok, fair point, although I would have thought the random calculation is quite fast anyway. Having said that, as all implementations follow the same approach, they will all incur the same "extra cost" of calculating the next random value, so it should cancel itself out.